lib/src/Base/Geom/MeshBoundaryFactory.cxx

```diff
@@ -0,0 +1,264 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Boundary extraction algorithm for meshes
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/MeshBoundaryFactory.hxx"
+#include <unordered_map>
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(MeshBoundaryFactory)
+static const Factory<MeshBoundaryFactory> Factory_MeshBoundaryFactory;
+
+
+/* Default constructor */
+MeshBoundaryFactory::MeshBoundaryFactory()
+  : PersistentObject()
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+MeshBoundaryFactory * MeshBoundaryFactory::clone() const
+{
+  return new MeshBoundaryFactory(*this);
+}
+
+/* String converter */
+String MeshBoundaryFactory::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << MeshBoundaryFactory::GetClassName();
+  return oss;
+}
+
+/* String converter */
+String MeshBoundaryFactory::__str__(const String & ) const
+{
+  return __repr__();
+}
+
+/* Here is the interface that all derived class must implement */
+
+
+namespace
+{
+struct FaceHasher
+{
+  UnsignedInteger operator()(const Indices &face) const
+  {
```

Comment:
maybe add a link to the method used, and a comment why we hash until the n-1 vertex

Reply:
done